public/components/getting_started/components/utils.tsx

```diff
@@ -20,16 +39,51 @@ const fetchAssets = async (tutorialId: string, assetFilter?: 'dashboards' | 'ind
   return responeData;
 };
 
-export const UploadAssets = async (tutorialId: string, mdsId: string, mdsLabel: string) => {
+export const UploadAssets = async (
+  tutorialId: string,
+  mdsId: string,
+  mdsLabel: string,
+  schema: ICollectorSchema[],
+  selectedIntegration: Array<EuiSelectableOption<CollectorOption>>
+) => {
   const { setToast } = useToast();
   const http = coreRefs.http;
 
+  let curIntegration: string | undefined;
+  if (selectedIntegration !== undefined) {
+    if (/^otel[A-Za-z]+$/i.test(selectedIntegration[0].value)) {
+      curIntegration = 'otel-services';
+    } else if (selectedIntegration[0].value === 'nginx') {
+      curIntegration = selectedIntegration[0].value;
+    }
+  }
+
   try {
+    // Auto-generate index templates based on the selected integration
+    let templates: ICollectorIndexTemplate[] = [];
+    if (curIntegration !== undefined) {
+      const indexTemplateMappings = await http!.get(
```

Comment:
Are we making this call on every selection or only once? we can cache this right.

Reply:
yes this is actually what I was considering also for an improvement later, but this is only triggered per 'create assets' button clicked, simply clicking on different cards won't trigger this call